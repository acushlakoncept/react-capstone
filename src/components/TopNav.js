import React from 'react';
import {
  Navbar, Nav, Form, FormControl, Button, Container,
} from 'react-bootstrap';

export default function TopNav() {
  return (
    <Navbar bg="dark" variant="dark">
      <Container>
        <Navbar.Brand href="#home">STOCKIST</Navbar.Brand>
        <Nav className="mr-auto">
          <Nav.Link href="#home">Home</Nav.Link>
          <Nav.Link href="#features">About</Nav.Link>
          <Nav.Link href="#pricing">Stock News</Nav.Link>
        </Nav>
        <Form inline>
          <FormControl type="text" placeholder="Search Stock" className="mr-sm-2" />
          <Button variant="outline-info">Search</Button>
        </Form>
      </Container>
    </Navbar>
  );
}
