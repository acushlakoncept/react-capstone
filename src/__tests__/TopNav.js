import React from 'react';
import renderer from 'react-test-renderer';
import TopNav from '../components/TopNav';

it('renders TopNav correctly', () => {
  const tree = renderer.create(<TopNav handleOnSelect={jest.fn()} />);
  expect(tree).toMatchSnapshot();
});
