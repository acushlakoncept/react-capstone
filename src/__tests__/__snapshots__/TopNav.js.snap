// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TopNav correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    STOCKIST
  </a>
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="mr-auto navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="/"
        href="/"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Home
      </a>
      <a
        className="nav-link"
        data-rb-event-key="/news"
        href="/news"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Stock News
      </a>
    </div>
    <form
      className="form-inline"
    >
      <select
        className="mr-sm-2 form-control"
        onChange={[MockFunction]}
      >
        <option
          value="active"
        >
          Most Active 
        </option>
        <option
          value="gainers"
        >
          Most Gainers 
        </option>
        <option
          value="losers"
        >
          Most Losers 
        </option>
      </select>
      <button
        className="btn btn-outline-success"
        disabled={false}
        type="button"
      >
        Filter by category
      </button>
    </form>
  </div>
</nav>
`;
